Source/Wikitivity.Agent/WikitivityAgentHelper.cs

```diff
@@ -5,43 +5,156 @@
 using kCura.Relativity.ImportAPI;
 using Newtonsoft.Json;
 using Relativity.API;
+using Relativity.Services.Objects;
+using Relativity.Services.Objects.DataContracts;
 using System;
 using System.Collections.Generic;
 using System.Data;
 using System.Net;
+using System.Threading.Tasks;
+using ObjectType = kCura.Relativity.Client.DTOs.ObjectType;
 
 namespace Wikitivity.Agent
 {
 	class WikitivityAgentHelper
 	{
 		public static string IAPIMessages = "";
-
-		public bool DeleteUploadedRequestsFromRelativity(IRSAPIClient proxy, int workspaceID, List<WikitivityUploadsAgent.SingleRequestObject> listOfCompletedUploads)
+		public static Guid WikitivityRDOGuid = new Guid("C6196733-E2A6-48F4-9443-37990972EBA3");
+		
+		public bool DeleteUploadedRequestsFromRelativityOM(IObjectManager proxy, int workspaceID, List<WikitivityUploadsAgent.SingleRequestObject> listOfCompletedUploads)
 		{
-			proxy.APIOptions.WorkspaceID = workspaceID;
-
 			bool success = false;
 			try
 			{
-				List<Document> DocsToDelete = new List<Document>();
+				List<RelativityObjectRef> listofObjectstoDelete = new List<RelativityObjectRef>();
 				foreach (var singleDoc in listOfCompletedUploads)
 				{
-					DocsToDelete.Add(new Document(singleDoc.ArtifactID));
+					listofObjectstoDelete.Add(new RelativityObjectRef() { ArtifactID = singleDoc.ArtifactID });
 				}
 
-				var DeleteResultSet = proxy.Repositories.Document.Delete(DocsToDelete);
-				if (DeleteResultSet.Success)
+				var deleteRequest = new MassDeleteByObjectIdentifiersRequest();
+				// Represents a list of RelativityObjects to be deleted.
+				deleteRequest.Objects = listofObjectstoDelete;
+				Task.Run(async () =>
+					{
+						await proxy.DeleteAsync(workspaceID, deleteRequest);
+					}).Wait();
+				return success = true;
+			}
+			catch (Exception ex)
+			{
+				throw new Exception("An error an occurred when deleting completed requests", ex);
+			}
+		}
+
+		public static async Task<List<int>> FindWikitivityCasesOM(IObjectManager proxy, int workspaceID)

```

Comment:
All Async methods should be suffixed with async like _FindWikitivityCasesAsync_. It's a good naming convention. There are couple other methods which don't follow the convention.